### ISSUE TITLE
CustomDataset does not convert bboxes to numpy format when a json file is used

### ISSUE THREAD

**Issue by eduramiba** [open-mmlab/mmdetection#387]
It casues python to try to multiply a standard array by a float.
Error trace:

```
Traceback (most recent call last):
  File "tools/train.py", line 90, in <module>
    main()
  File "tools/train.py", line 86, in main
    logger=logger)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmdet/apis/train.py", line 59, in train_detector
    _non_dist_train(model, dataset, cfg, validate=validate)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmdet/apis/train.py", line 121, in _non_dist_train
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmcv/runner/runner.py", line 355, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmcv/runner/runner.py", line 257, in train
    for i, data_batch in enumerate(data_loader):
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 637, in __next__
    return self._process_next_batch(batch)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 658, in _process_next_batch
    raise batch.exc_type(batch.exc_msg)
TypeError: Traceback (most recent call last):
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 138, in _worker_loop
    samples = collate_fn([dataset[i] for i in batch_indices])
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 138, in <listcomp>
    samples = collate_fn([dataset[i] for i in batch_indices])
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmdet/datasets/custom.py", line 154, in __getitem__
    data = self.prepare_train_img(idx)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmdet/datasets/custom.py", line 209, in prepare_train_img
    flip)
  File "/home/paperspace/miniconda3/envs/pytorch/lib/python3.7/site-packages/mmdet/datasets/transforms.py", line 76, in __call__
    gt_bboxes = bboxes * scale_factor
TypeError: can't multiply sequence by non-int of type 'float'
```

**Comment by hellock** [comment]
Usually we use pickle files for annotation, where the bboxes are already numpy array.
You can open a PR to fix the support for json files.

**Comment by eduramiba** [comment]
@hellock Thank you for the help. Do you think you should remove support for JSON then? Indeed I also will need to convert my datataset to COCO format, given evaluation won't also work with my (patched) JSON `CustomDataset` due to issue #145 too.

I think incomplete support should be removed for dataset types that cannot be trained and evaluated correctly.

Thanks!

**Comment by hellock** [comment]
It depends. If you want to evaluate with the coco api, then you have to convert the annotation files to the coco format. Otherwise you can just use a simple pickle or json file, just as illustrated in CustomDataset. For such datasets, we also provide support for evaluation, but with PASCAL VOC like metrics.
